troposphere/static/js/components/modals/image_version/ImageSelect.react.js

```diff
@@ -11,7 +11,12 @@ export default React.createClass({
     },
 
     handleChange: function(e){
-      this.props.onChange(e.target.value)
+      let image_id = e.target.value;
+      let username = stores.ProfileStore.get().get('username'),
+          image = stores.ImageStore.fetchWhere({
+          created_by__username: username
+      }).get(image_id);
```

Comment:
Just a continuation ... I think the `username = ... ` was all part of the same expression when I looked at it initially.